src/testFixtures/java/org/opensearch/knn/KNNRestTestCase.java

```diff
@@ -801,6 +820,129 @@ public void validateKNNSearch(String testIndex, String testField, int dimension,
         }
     }
 
+    protected Settings getKNNIndexCustomLegacyFieldMappingSettings(SpaceType spaceType, Integer m, Integer ef_construction) {
+        return Settings.builder()
+            .put(NUMBER_OF_SHARDS, 1)
+            .put(NUMBER_OF_REPLICAS, 0)
+            .put(INDEX_KNN, true)
+            .put(KNNSettings.KNN_SPACE_TYPE, spaceType.getValue())
+            .put(KNNSettings.KNN_ALGO_PARAM_M, m)
+            .put(KNNSettings.KNN_ALGO_PARAM_EF_CONSTRUCTION, ef_construction)
+            .build();
+    }
+
+    public String createKnnIndexMethodFieldMapping(String fieldName, Integer dimensions) throws IOException {
+        return Strings.toString(
+            XContentFactory.jsonBuilder()
+                .startObject()
+                .startObject(PROPERTIES)
+                .startObject(fieldName)
+                .field(VECTOR_TYPE, KNN_VECTOR)
+                .field(DIMENSION, dimensions.toString())
+                .startObject(KNN_METHOD)
+                .field(NAME, METHOD_HNSW)
+                .endObject()
+                .endObject()
+                .endObject()
+                .endObject()
+        );
+    }
+
+    public String createKnnIndexCustomMethodFieldMapping(
+        String fieldName,
+        Integer dimensions,
+        SpaceType spaceType,
+        String engine,
+        Integer m,
+        Integer ef_construction
+    ) throws IOException {
+        return Strings.toString(
+            XContentFactory.jsonBuilder()
+                .startObject()
+                .startObject(PROPERTIES)
+                .startObject(fieldName)
+                .field(VECTOR_TYPE, KNN_VECTOR)
+                .field(DIMENSION, dimensions.toString())
+                .startObject(KNN_METHOD)
+                .field(NAME, METHOD_HNSW)
+                .field(METHOD_PARAMETER_SPACE_TYPE, spaceType.getValue())
+                .field(KNN_ENGINE, engine)
+                .startObject(PARAMETERS)
+                .field(METHOD_PARAMETER_EF_CONSTRUCTION, ef_construction)
+                .field(METHOD_PARAMETER_M, m)
+                .endObject()
+                .endObject()
+                .endObject()
+                .endObject()
+                .endObject()
+        );
+    }
+
+    // Default KNN script score settings
+    protected Settings getKNNScriptScoreSettings() {
+        return Settings.builder().put(NUMBER_OF_SHARDS, 1).put(NUMBER_OF_REPLICAS, 0).put(INDEX_KNN, false).build();
+    }
+
+    // Validate script score search for these space_types : {"l2", "l1", "linf"}
```

Comment:
Space type seems like an argument. Im not sure it belongs in the comment.

Reply:
That method supports those 3 space_types. Though we are handling it, I thought adding a comment would be helpful